mobxjs/mobx-state-tree
# Convert to bun:test for our tests, remove codecov

## What does this PR do and why?

This PR will convert our Jest test to [bun test](https://bun.sh/docs/cli/test). This offers us a few advantages:

1. Huge improvement in speed (locally, I see ~2-3x improvement)
2. Native support for TypeScript, so no need for `ts-jest`
3. Without the need for `ts-jest` and `jest` itself, we reduce our `devDependencies` as well
4. Since we [have moved over to Bun for dependency management and script running](https://github.com/mobxjs/mobx-state-tree/pull/2148), this makes our development tooling more consistent overall.
5. This also removes [codecov](https://www.npmjs.com/package/codecov), partially because we didn't take the time to implement it with Bun, but also because that package is deprecated, and I don't think our token has been working for some time. We discussed this in the maintainer chat and agreed.
6. Fixes a missing `MstError` from a rebase

## Steps to validate locally

1. Make sure you have [latest Bun installed](https://bun.sh/docs/installation)
7. Pull down this branch
8. Run `bun install`
9. Then run `bun test:all` - tests should pass.
10. If you time it with `time bun test:all`, you can measure its runtime. Here's what I got on my 2020 Macbook Pro with Intel i7/32 GB RAM:

```
bun test:all  14.55s user 6.84s system 100% cpu 21.302 total
```

Compare this to the `yarn test:all` run in `master`, which reports:

```
Ran all test suites.
✨  Done in 75.39s.
```

Ran it with `time yarn test:all` and got slightly better results but with a lot more CPU usage:

```
yarn test:all  178.77s user 24.75s system 432% cpu 47.026 total
```
